Docs for composer create-project using vendors (#29)
packagist/private-packagist-docs: docs/setup-vendor.md

# Getting started with Private Packagist vendors on a Composer project

> You can start using Private Packagist from a vendor in either a new or an existing Composer project.

Reviewer: This needs to explain what vendor is and why someone would want to run `composer init` or `composer create-project`. Similar to how the setup guide explains how to create a new Composer project with Private Packagist and then goes on to list the two options.

To make it clearer why someone might not need to run either of those two commands it also might make sense to show how to just add setup auth, add the repo to an existing composer.json and add a single package. As this might be how most vendor customers will use Private Packagist.

Author: ok, I tried restructuring this and added a section for installing a single package.

> # Vendor's Customers using Private Packagist in a Composer project

Reviewer 2: ```suggestion
# Private Packagist for Vendors: Using Private Packages in Customer Projects
```

Reviewer 2: This has not been applied yet?

Author: ok applied now and double checked the rest.

> # Getting started with Private Packagist for Vendors in a Composer project

Reviewer 2: Think of a more fitting headline in this context.

> Using Private Packagist for Vendors, customers install packages with Composer using a unique URL and an authentication token.

Reviewer 2: whose customers? Maybe say "your customers can install ..."

> Using Private Packagist for Vendors, your customers can install packages with Composer using a unique URL and an authentication token.

Reviewer 2: ```suggestion
Using Private Packagist for Vendors, your customers can install packages with Composer using a unique URL and an authentication token. You can create customers and manage their package access customers on the vendor page in your organization.
```

> Customers can start using your packages in either an existing or a new Composer project.

Reviewer 2: ```suggestion
Customers can use your packages in either an existing or a new Composer project.
```

> Customers can start using Private Packagist in either an existing or a new Composer project.

Reviewer 2: They don't want to start using Private Packagist, they want to start using the vendor's packages.

> ## Adding a Package to an existing Composer project

Reviewer 2: ```suggestion
## Adding a package to an existing Composer project
```

> To install a package named "acme-company/api" into your existing project, The following steps are required:

Reviewer 2: "your" refers to the customer here, but this is supposed to be written for vendors, please review the article again and let me know when you've updated it to be directed at vendors rather than customers.

Author: ok, I updated it and made sure the rest of this article is directed at vendors.

> To install a package named "acme-company/api" into your existing project, you need to follow the steps below:

Reviewer 2: "You" in this case is the vendor, not the customer, I'll stop the review here. Please re-read the article, make sure the entire article is directed at vendors, not customers, then ping me to review again, thanks!

Author: ok updated and made sure it's directed towards vendors.

1. Configure authentication. The command to set an authentication token is listed on the customer Composer information page.
2. Add the customer repository in the project's composer.json. The customer repository URL is listed on the customer Composer information page. The project's composer.json should now have a new entry:
    ```
    "repositories": [
        {"type": "composer", "url": "https://acme.repo.packagist.com/customer/"}
    ]
    ```
3. Install the package by running `composer require acme-company/api`
           
![Customer Composer Information](/Resources/public/img/docs/customer-composer-information.png)

> ## Starting a new Composer Project

Reviewer 2: ```suggestion
## Starting a new Composer project
```

To set up a new project, your customers can initialize the project with the Customer's Composer repository URL or use a template package.

### composer init
To initialize a new Composer project the following steps are required:

1. Configure authentication. The command to set an authentication token is listed on the customer Composer information page.
2. Run `composer init --name=acme/website --repository=https://acme.repo.packagist.com/customer/` . The customer repository URL is listed on the customer Composer information page. 

### composer create-project

> To initialize a project using a template package named `acme/website`, then these steps are required:

Reviewer 2: ```suggestion
If you wish to provide a template package for new projects to your customers, the following steps show how to initialize a project from an example template package named `acme/website`. Your template project should not contain a Private Packagist repository URL in its composer.json, so the customer's own URL can be added while creating a new project.
```

> 1. Configure authentication. The command to set an authentication token is listed on the customer Composer information page.

Reviewer 2: ```suggestion
1. Configure authentication: The command to set an authentication token is listed under the vendor tab on the customer's "Composer Instructions" page.
```

> 2. Run `composer create-project acme/website --add-repository --repository=https://acme.repo.packagist.com/customer/` . The customer repository URL is listed on the customer Composer information page. 

Reviewer 2: ```suggestion
2. Run `composer create-project acme/website --add-repository --repository=https://acme.repo.packagist.com/customer/` . The customer repository URL is listed under the vendor tab on the customer's "Composer Instructions" page. 
```

> Once the command is finished, the created composer.json file will contain the added repositories:

Reviewer 2: ```suggestion
Once the command is finished, the created composer.json file will contain the customer's Private Packagist repository URL:
```

```
"repositories": [
    {"type": "composer", "url": "https://acme.repo.packagist.com/customer/"}
]
```

> *Important Note:* While using the `add-repository` flag and in case a composer.lock file is present, the lock file will be deleted and an update will be run instead of install.

Reviewer 2: ```suggestion
*Important Note:* If there is a composer.lock in your template project, the `add-repository` option will delete the lock file and run an update to generate a new one using the customer's personalized repository URL.
```

# Getting started with Private Packagist for vendors on a Composer project

Using Private Packagist, vendors let their customers install packages with Composer using a unique URL and an authentication token.

As a vendor's customer, you can start using Private Packagist in either an existing or a new Composer project.

## Adding a Package from a vendor to an existing Composer project

> Let's suppose your company bought a package named `acme-company/api` from a vendor named Acme Company. You'd like to install the package in your existing project.

Reviewer: I am confused. Didn't we say that the documentation is for the package vendor with the goal that they have enough info to explain to their customers how they can install the packages they bought? This reads like it addresses directly the vendor customer.

Author: What I wanted to acheive with this, is to show the customer how they can install a package from a vendor. In this case, the customer bought a package from a vendor, and the vendor is trying to explain how to install a package. That's why it's addressing directly the vendor customer. I'm not sure if there is a better way here?

Reviewer 2: @wissem The goal of our documentation is to inform our customers, who are vendors, what information to give to their customers. This documentation will not be read by the customers of packages themselves.

Author: ok, this should be adjusted now.

> # Using Private Packagist with the vendor addon in a Composer project

Reviewer: We don't call them add-ons anymore

To start a new project using Composer and Private Packagist with the [vendor](https://packagist.com/vendors) addon, you can use the following command:

1. Configure authentication on your machine. The command to set an authentication token is listed on the customer Composer information page.

> 2. Run `composer create-project acme/website --add-repository --repository=http://acme-company.repo.packagist.com/customer1/` You can find the customer repository URL on the customer Composer information page. 

Reviewer: Is there a reason the docs/setup.md has a version but the example here doesn't?
